substrate/command_line/arguments

```diff
@@ -58,7 +60,8 @@ namespace substrate::commandLine
 		arguments_t &operator =(const arguments_t &arguments) noexcept;
 		arguments_t &operator =(arguments_t &&arguments) noexcept;
 
-		[[nodiscard]] bool parseFrom(internal::tokeniser_t &lexer, const options_t &options);
+		[[nodiscard]] bool parseFrom(internal::tokeniser_t &lexer, const options_t &options,
+			optionsVisited_t globalOptions);
```

Comment:
This should enable you to drop all the `{}` initializations earlier on. Unless you want to ensure a future self thinks twice about the map to be passed here?

```suggestion
			optionsVisited_t globalOptions = {});
```

Reply:
That's an interesting and good question to ask - think that, because it's by-val specifically (so that we don't accidentally back-propagate global options from inner to outer parsing contexts), we're going to stick with the way it's written so we have to think twice.